core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -161,42 +184,22 @@
       // invalid scheme. Just fall through and will try "dns:///foo.googleapis.com:8080"
```

Comment:
`foo.googleapis.com:8080` won't throw, but `127.0.0.1:8080` will. The first character of the scheme must not be a digit.